packages/react-client/src/components/coffee/CoffeeIcon.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { SyntheticEvent, useState } from 'react';
+
+type CoffeeIconProps = {
+  index: number;
+  value: number;
+  hoverValue: number;
+  onValueChange: (newValue: number) => void;
+  onHoverValueChange: (newHoverValue: number) => void;
+}
+
+const CoffeeIcon = (event: CoffeeIconProps) => {
+  const [hover, setHover] = useState(false);
+  const selectedColor = '#4e342e';
+  const notSelectedColor = '#bdbdbd';
+  const hoverColor = '#1565c0';
+  let fillColor: string;
+
+  if (hover) {
+    fillColor = hoverColor;
+  } else {
+    fillColor = event.index < event.hoverValue ? hoverColor : notSelectedColor;
+  }
+  if (event.hoverValue === 0) {
+    fillColor = event.index <= event.value ? selectedColor : notSelectedColor;
+  }
+
+  const toggleHover = () => {
+    setHover(!hover);
+    event.onHoverValueChange(!hover ? event.index : 0);
+  }
+
+  const handleClick = (e: SyntheticEvent) => {
+    event.onValueChange(event.index);
```

Comment:
where is `event` coming from? Do you mean `e`? but maybe it should be `clickEvent` ;)

Reply:
ahhhh, I see now, it's coming from way up top.

how about this instead of calling it `event`

`const CoffeeIcon = (props: CoffeeIconProps)`